packages/components/src/modal/Modal.tsx

```diff
@@ -0,0 +1,183 @@
+import classNames from 'classnames';
+import React, {
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import ThemeExport from '../ThemeExport';
+
+interface ModalProps {
+  className?: string;
+  children?: ReactNode;
+  role?: string;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  size?: 'sm' | 'lg' | 'xl' | undefined;
+  onOpened?: () => void;
+  onClosed?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}
+
+const Modal = ({
+  className = 'theme-bg-light',
+  children,
+  role = 'role',
+  keyboard = true,
+  isOpen = false,
+  centered = false,
+  size,
+  onOpened,
+  onClosed,
+  toggle,
+  'data-testid': dataTestId,
+}: ModalProps): ReactElement => {
+  const [isPortalOpen, setIsPortalOpen] = useState(false);
+
+  const element = useRef<HTMLElement>();
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case 'Escape':
+          if (toggle) {
+            toggle();
+          }
```

Comment:
Can write this on one line with optionals:
```suggestion
          toggle?.();
```